From ea77273ddc4cf48684bbfdd3d46a9f43dfe7351c Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Thu, 30 May 2019 08:01:42 +0300
Subject: [PATCH] [SW-0] pci: sync earlier with BTL PCIe image

Patch 5bd5c00 [SW-0] pcie: add log of image id (2019-05-23 - Noam Camus)
Is causing a race condition on the UART between PCIe image and the
Preboot.
Solution is to sync with PCIe image before any usage of UART.

This is possible since:
In GOYA the PRSTn is connected to rst_n via CPLD. So this check is
irrelevant (once we reach pre-boot, prstn is always de-asserted)
For GAUDI, we can move the check to an earlier stage since the pre-boot
does not control the PCIe- all is done via the BTL instructions.

Change-Id: Id8cf61295efba60cc03fcc29a7487d87643debc3
---

diff --git a/common/preboot.c b/common/preboot.c
index bfe7083..d476346 100644
--- a/common/preboot.c
+++ b/common/preboot.c
@@ -169,11 +169,11 @@
 	regmap_writel(CPU_BOOT_STATUS_IN_PREBOOT, get_cpu_boot_status_reg());
 
 	hl_jtag_init();
+	preboot_sync_btl();
 	hl_uart_init();
 	preboot_version();
 	preboot_pcie_init();
 	preboot_delay();
-	preboot_sync_btl();
 	hl_pll_init();
 	/* reinit uart after PLLs configured */
 	hl_uart_init();
